src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -470,6 +470,11 @@ protected virtual void Initialize()
                 var functionMetadata = ReadFunctionMetadata(ScriptConfig, TraceWriter, _startupLogger, FunctionErrors, _settingsManager);
                 var usedBindingTypes = DiscoverBindingTypes(functionMetadata);
 
+                if (!ScriptConfig.LazyLoadExtensions)
```

Comment:
I wanted an approach that unified the paths as much as possible and exercised the lazy loading machinery even in the mainline case.